### PR TITLE
Update createTextureView API to include swizzling, implement on Metal

### PR DESCRIPTION
**Pull request by bejado**

### CODE REVIEW

**Review by pixelflinger — CHANGES_REQUESTED** [review]
I like this!

File: filament/backend/test/test_FeedbackLoops.cpp [hunk 1]
```diff
@@ -230,7 +230,9 @@ TEST_F(BackendTest, FeedbackLoops) {
                 params.viewport.width = kTexWidth >> targetLevel;
                 params.viewport.height = kTexHeight >> targetLevel;
 
-                auto textureView = api.createTextureView(texture, sourceLevel, 1);
+                auto textureView = api.createTextureView(texture, sourceLevel, 1,
+                        backend::TextureSwizzle::CHANNEL_0, backend::TextureSwizzle::CHANNEL_1,
+                        backend::TextureSwizzle::CHANNEL_2, backend::TextureSwizzle::CHANNEL_3);
```
[the remark this answers is not in the document]
Comment (bejado) [reply filament/backend/test/test_FeedbackLoops.cpp:235]: Yeah I was thinking about that. `createTextureViewSwizzle` would certainly be more ergonomic to use. Let me update it

**Review by pixelflinger — APPROVED** [review]
Looks good to me. Do you intend to implement the vk/gl sides, or should we do it?

**Comment by bejado** [comment]
> Looks good to me. Do you intend to implement the vk/gl sides, or should we do it?

If you can do the GL side that would be great. I started to look into it but I got confused.

@poweifeng do you have bandwidth to implement the VK side? If not I'm happy to try to implement it myself.

**Comment by poweifeng** [comment]
> > Looks good to me. Do you intend to implement the vk/gl sides, or should we do it?
> 
> If you can do the GL side that would be great. I started to look into it but I got confused.
> 
> @poweifeng do you have bandwidth to implement the VK side? If not I'm happy to try to implement it myself.

Sure, I can give it a try later this week.

**Pull request merged**